gradle-baseline-java/src/main/java/com/palantir/baseline/tasks/CheckJUnitDependencies.java

```diff
@@ -16,211 +16,194 @@
 
 package com.palantir.baseline.tasks;
 
-import static java.util.stream.Collectors.toList;
-
 import com.google.common.base.Preconditions;
-import com.palantir.baseline.plugins.BaselineTesting;
-import com.palantir.baseline.util.VersionUtils;
-import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
 import java.nio.file.Files;
+import java.util.ArrayList;
 import java.util.List;
-import java.util.Optional;
 import java.util.Set;
 import java.util.function.Predicate;
 import java.util.stream.Stream;
 import org.gradle.api.DefaultTask;
-import org.gradle.api.artifacts.Configuration;
 import org.gradle.api.artifacts.ModuleVersionIdentifier;
 import org.gradle.api.artifacts.result.ResolvedComponentResult;
 import org.gradle.api.plugins.JavaPluginExtension;
-import org.gradle.api.provider.Provider;
-import org.gradle.api.tasks.Classpath;
+import org.gradle.api.provider.ListProperty;
+import org.gradle.api.tasks.Input;
 import org.gradle.api.tasks.SourceSet;
 import org.gradle.api.tasks.TaskAction;
 import org.gradle.api.tasks.testing.Test;
+import org.gradle.api.tasks.testing.TestFrameworkOptions;
+import org.gradle.api.tasks.testing.junitplatform.JUnitPlatformOptions;
 
-public class CheckJUnitDependencies extends DefaultTask {
+public abstract class CheckJUnitDependencies extends DefaultTask {
+    @Input
+    public abstract ListProperty<String> getErrorMessages();
 
     public CheckJUnitDependencies() {
         setGroup("Verification");
         setDescription("Ensures the correct JUnit4/5 dependencies are present, otherwise tests may silently not run");
-        getOutputs().upToDateWhen(_task -> true);
```

Comment:
Tasks with no outputs are never up-to-date

Reply:
Err sorry, I did not intend to delete this.